pyogrio/geopandas.py

```diff
@@ -552,6 +568,74 @@ def write_dataframe(
         else:
             crs = geometry.crs.to_wkt(WktVersion.WKT1_GDAL)
 
+    if use_arrow:
+        import pyarrow as pa
+        from pyogrio.raw import write_arrow
+
+        if geometry_column is not None:
```

Comment:
We should raise an exception if geometry_column is None, otherwise we get an error that doesn't clearly indicate that

```
>>> write_dataframe(pd.DataFrame(df.drop(columns=['geometry'])),'/tmp/test.dbf', use_arrow=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../pyogrio/pyogrio/geopandas.py", line 609, in write_dataframe
    geom_field = table.schema.field(geometry_column)
  File "pyarrow/types.pxi", line 2818, in pyarrow.lib.Schema.field
TypeError: Index must either be string or integer
```